bundles/CoreBundle/src/Migrations/Version20230616085142.php

```diff
@@ -51,10 +52,13 @@ public function up(Schema $schema): void
             $tableName = current($table);
             $metaDataTable = $schema->getTable($tableName);
             $foreignKeyName = AbstractDao::getForeignKeyName($tableName, self::ID_COLUMN);
+            $foreignKeyNameWithOPrefix = AbstractDao::getForeignKeyName($tableName, self::O_PREFIX . self::ID_COLUMN);
 
             if (!$metaDataTable->hasColumn(self::AUTO_ID)) {
                 if ($recreateForeignKey = $metaDataTable->hasForeignKey($foreignKeyName)) {
                     $this->addSql('ALTER TABLE `' . $tableName . '` DROP FOREIGN KEY `' . $foreignKeyName . '`');
+                } elseif ($recreateForeignKey = $metaDataTable->hasForeignKey($foreignKeyNameWithOPrefix)) {
```

Comment:
Fine for more idempotency, but this suggest that this migration (with a more "in-the-past" timestamp) didn't run properly?

https://github.com/pimcore/pimcore/blob/40890a3c30481fec0d15dbd0b550b1865a55d8fc/bundles/CoreBundle/src/Migrations/Version20221003115124.php#L90-L98

In other words, if that Version20221003115124 ran successfully, Version20230616085142 should be fine already to stop considering the `o_` fallback, or am i missing something?

Reply:
Ah, maybe i got it, it's the name of the foreign key still containing `o_id`, even though the column reference is to `id`

Reply:
Exactly, the name of the foreign key does not get updated. 

Although minor, this also adds the `ON UPDATE NO ACTION` to the foreign key constraint.